zoo/analytics/tasks/utils.py

```diff
@@ -59,3 +61,64 @@ def resolve_unknown_format(version: str) -> str:
         else:
             break
     return ".".join(temp_version)
+
+
+class DockerImageId:
+    """Represents parsed docker image id.
+
+    TODO: move this to some more appropriate OSS package.
+    """
+
+    # pylint: disable=too-many-instance-attributes # Ignore PyCommentedCodeBear
+
+    def __init__(self, image: str):  # Ignore RadonBear
+        self.full_image_name = image
+
+        self.registry = None
+        self.port = None
+        self.username = None
+        self.repository_name = None
+        self.namespaces = None
+        self.tag = None
+        self.version = None
+        self.os = None
+        self.os_version = None
+        self.alias = None
+
+        if "/" in image:
```

Comment:
as ai read that parsing, i think the properties should be here. It would be nice to have 1 line comment to what it does.